src/tools/generate/generate_figure_ids.js

```diff
@@ -1,8 +1,16 @@
 const generate_figure_ids = (html) => {
-  const re = /<figure>|<figure markdown>/gi;
+  const re = /<figure>|<figure data-markdown>/gi;
 
   let i = 1;
-  return html.replace(re, () => `<figure id='fig-${i++}'>`);
+  html = html.replace(re, () => `<figure id='fig-${i++}'>`);
+
+  //Some of our 2019 chapters (markdown and performance) had different fig ids
+  const figcaption_regex = /<figcaption(.*?)>(.*?)([0-9]+)\./gi;
+  if (html.includes('<figure id="fig1"')) {
+    return html.replace(figcaption_regex, "<figcaption$1><a href=\"#fig$3\" class=\"anchor-link\">$2 $3.</a>");
```

Comment:
Nit: use single quotes for strings

Reply:
Fixed. Force of habit due to my Perl days where single quotes and double quotes are different (variables are not evaluated in single quotes in Perl). Will try to remember JavaScript is different! 😀